code/modules/paperwork/photocopier.dm

```diff
@@ -276,47 +140,320 @@
+
+		if(!camera)
+			return
+		var/datum/picture/selection = camera.selectpicture()
+		if(!selection)
+			return
+
+		playsound(loc, 'sound/goonstation/machines/printer_dotmatrix.ogg', 50, 1)
+		var/obj/item/photo/p = new /obj/item/photo (src.loc)
+		p.construct(selection)
+		if(p.desc == "")
+			p.desc += "Copied by [tempAI.name]"
+		else
+			p.desc += " - Copied by [tempAI.name]"
+		toner -= 5
+		SStgui.update_uis(src)
+		sleep(15)
+
+/obj/machinery/photocopier/attackby(obj/item/O as obj, mob/user as mob, params)
+	if(istype(O, /obj/item/paper) || istype(O, /obj/item/photo) || istype(O, /obj/item/paper_bundle))
+		if(!copyitem)
+			user.drop_item()
+			copyitem = O
+			O.forceMove(src)
+			to_chat(user, "<span class='notice'>You insert \the [O] into \the [src].</span>")
+			flick(insert_anim, src)
+			SStgui.update_uis(src)
+		else
+			to_chat(user, "<span class='notice'>There is already something in \the [src].</span>")
+	else if(istype(O, /obj/item/toner))
+		if(toner <= 10) //allow replacing when low toner is affecting the print darkness
+			user.drop_item()
+			to_chat(user, "<span class='notice'>You insert the toner cartridge into \the [src].</span>")
+			var/obj/item/toner/T = O
+			toner += T.toner_amount
+			qdel(O)
+		else
+			to_chat(user, "<span class='notice'>This cartridge is not yet ready for replacement! Use up the rest of the toner.</span>")
+	else if(istype(O, /obj/item/folder))
+		if(!folder) //allow replacing when low toner is affecting the print darkness
+			user.drop_item()
+			to_chat(user, "<span class='notice'>You slide the [O] into \the [src].</span>")
+			folder = O
+			O.forceMove(src)
+			SStgui.update_uis(src)
+		else
+			to_chat(user, "<span class='notice'>This cartridge is not yet ready for replacement! Use up the rest of the toner.</span>")
+	else if(istype(O, /obj/item/grab)) //For ass-copying.
+		var/obj/item/grab/G = O
+		if(ismob(G.affecting) && G.affecting != M)
+			var/mob/GM = G.affecting
+			visible_message("<span class='warning'>[usr] drags [GM.name] onto the photocopier!</span>")
+			GM.forceMove(get_turf(src))
+			M = GM
+			if(copyitem)
+				copyitem.forceMove(get_turf(src))
+				copyitem = null
+			SStgui.update_uis(src)
+	else
+		return ..()
+
+/obj/machinery/photocopier/wrench_act(mob/user, obj/item/I)
+	. = TRUE
+	default_unfasten_wrench(user, I)
+
 /obj/machinery/photocopier/obj_break(damage_flag)
 	if(!(flags & NODECONSTRUCT))
 		if(toner > 0)
 			new /obj/effect/decal/cleanable/blood/oil(get_turf(src))
 			toner = 0
 
 /obj/machinery/photocopier/MouseDrop_T(mob/target, mob/user)
-	if(!istype(target) || target.buckled || get_dist(user, src) > 1 || get_dist(user, target) > 1 || user.stat || istype(user, /mob/living/silicon/ai) || target == ass)
+	if(!istype(target) || target.buckled || get_dist(user, src) > 1 || get_dist(user, target) > 1 || user.stat || istype(user, /mob/living/silicon/ai) || istype(src, /obj/machinery/photocopier/faxmachine))
```

Comment:
This should not be done like this. Just override the fax machine's proc (I alerady mentioned this in one of my earlier reviews though).
```suggestion
	if(!istype(target) || target.buckled || get_dist(user, src) > 1 || get_dist(user, target) > 1 || user.stat || istype(user, /mob/living/silicon/ai))
```